File: PlantHereNodeJs/Core/PlantHere.Aplication/RequestResponseModels/Product/Commands/UpdateProduct/UpdateProductCommand.js
const Guid = require("../../../../Services/GuidService")

class UpdateProductCommand {

    constructor(id,name,description,price,sellerId,categoryId,discount) {
        this.Id = id,
        this.Name = name,
        this.Description = description,
        this.Price = price,
        this.SellerId = sellerId,
        this.CategoryId = categoryId,
        this.Discount = discount,
        this.UpdatedDate = Guid.dateNow();  
    }
}

module.exports = {UpdateProductCommand}

File: PlantHere/PlantHere.Application/CQRS/Product/Commands/UpdateProduct/UpdateProductCommand.cs
using MediatR;
using PlantHere.Application.CQRS.Base;

namespace PlantHere.Application.CQRS.Product.Commands.UpdateProduct
{
    public class UpdateProductCommand : CommandBase<Unit>
    {
        public int Id { get; set; }

        public string? Name { get; set; }

        public string? Description { get; set; }

        public int Stock { get; set; }

        public int Price { get; set; }

        public int Discount { get; set; }

        public int CategoryId { get; set; }

        public UpdateProductCommand(int id, string? name, string? description, int stock, int price, int discount, int categoryId)
        {
            Id = id;
            Name = name;
            Description = description;
            Stock = stock;
            Price = price;
            Discount = discount;
            CategoryId = categoryId;
        }
    }
}
